src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseTestCase.java

```diff
@@ -178,4 +180,17 @@ public String toString() {
         return "ProgrammingExerciseTestCase{" + "id=" + getId() + ", testName='" + testName + '\'' + ", weight=" + weight + ", active=" + active + ", visibility=" + visibility
                 + ", bonusMultiplier=" + bonusMultiplier + ", bonusPoints=" + bonusPoints + '}';
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        ProgrammingExerciseTestCase that = (ProgrammingExerciseTestCase) o;
+        return isSameTestCase(that);
+    }
```

Comment:
```suggestion
    public boolean equals(Object other) {
        if (this == other) {
            return true;
        }

        if (other instanceof ProgrammingExerciseTestCase otherTestCase) {
            return isSameTestCase(otherTestCase);
        } else {
            return false;
        }
    }
```
The current implementation would fail to cast `o` to a `ProgrammingExerciseTestCase` if any other object type is given as a parameter.